tests/phpunit/tests/db.php

```diff
@@ -1717,6 +1729,36 @@ public function data_prepare_with_placeholders() {
 				false,
 				"'hello' 'foo##'",
 			),
+			array(
+				'SELECT * FROM %i WHERE %i = %d;',
+				array( 'my_table', 'my_field', 321 ),
+				false,
+				'SELECT * FROM `my_table` WHERE `my_field` = 321;',
+			),
+			array(
+				'WHERE %i = %d;',
+				array( 'evil_`_field', 321 ),
+				false,
+				'WHERE `evil_``_field` = 321;', // To quote the identifier itself, then you need to double the character, e.g. `a``b`.
+			),
+			array(
+				'WHERE \'%i\' = 1 AND "%i" = 2 AND `%i` = 3 AND %15i = 4',
+				array( 'my_field1', 'my_field2', 'my_field3', 'my_field4' ),
+				false,
+				'WHERE \'`my_field1`\' = 1 AND "`my_field2`" = 2 AND ``my_field3`` = 3 AND `      my_field4` = 4', // Does not remove any existing quotes, always adds it's own (safer).
+			),
+			array(
+				'WHERE %i LIKE %1$s LIMIT 1',
+				array( 'field -- ', false ),
+				true, // Incorrect usage.
+				null, // Should be rejected, otherwise the `%1$s` could use Identifier escaping, e.g. 'WHERE `field -- ` LIKE field --  LIMIT 1' (thanks @vortfu).
+			),
+			array(
+				'WHERE %i LIKE %s LIMIT 1',
+				array( "field' -- ", "field' -- " ),
+				false,
+				"WHERE `field' -- ` LIKE 'field\' -- ' LIMIT 1", // In contrast to the above, Identifier vs String escaping is used.
+			),
```

Comment:
Also could use tests with and without `$unsafe_unquoted_parameters` set.

Reply:
Good point, I've added some basic tests with `test_allow_unsafe_unquoted_parameters`, with the [existing tests](https://github.com/WordPress/wordpress-develop/blob/26bb0d7479e1dd08dd982513ee6bee94f4c6f13d/tests/phpunit/tests/db.php#L1642) continuing to check the old behaviour has not changed.

This also helped me notice that getting `argnum` from the format string was incorrect, and always used 1 (no idea how I missed that); so I've also updated this test use parameter 2:

```diff
- 'WHERE %i LIKE %1$s LIMIT 1',
+ 'WHERE id = %d AND %i LIKE %2$s LIMIT 1',
```

Reply:
This was updated in commit [1cc5799](https://github.com/WordPress/wordpress-develop/pull/2192/commits/1cc5799df9026f98b85ddf8ea4319d4ecfa372b9)